packages/core/contracts/financial-templates/common/financial-product-libraries/long-short-pair-libraries/SuccessTokenLongShortPairFinancialProductLibrary.sol

```diff
@@ -58,16 +63,23 @@ contract CoveredCallLongShortPairFinancialProductLibrary is LongShortPairFinanci
         require(contractStrikePrice != 0, "Params not set for calling LSP");
 
         // If the expiry price is less than the strike price then the long options expire worthless (out of the money).
+        // In this case, return of value of 50% (half of collateral goes to long)
         // Note we do not consider negative expiry prices in this call option implementation.
         if (expiryPrice < 0 || uint256(expiryPrice) < contractStrikePrice)
-            return FixedPoint.fromUnscaledUint(0).rawValue;
+            return FixedPoint.fromUnscaledUint(0.5).rawValue;
 
-        // Else, token expires to be worth the fraction of a collateral token that's in the money. eg if ETH is $3500
-        // and strike is $3000, long token is redeemable for (3500-3000)/3500 = 0.143 WETH which is worth $500 and the
-        // short token is worth the remaining 0.8. This is strictly < 1, tending to 1 as the expiry tends to infinity.
+        // Else, token expires to be worth the 0.5 of the collateral plus 0.5 * the fraction of a collateral token 
+        // that's in the money. 
+        // eg if SUSHI is $30 and strike is $20, long token is redeemable for 0.5 + 0.5*(30-20)/30 = 0.6667% which if the
+        // collateralPerPair is 2, is worth 1.3333 $SUSHI, which is worth $40 if 1 $SUSHI is worth $30.
+        // This return value is strictly < 1, tending to 1 as the expiryPrice tends to infinity.
         return
-            (FixedPoint.Unsigned(uint256(expiryPrice)).sub(FixedPoint.Unsigned(contractStrikePrice)))
-                .div(FixedPoint.Unsigned(uint256(expiryPrice)))
-                .rawValue;
+            (FixedPoint.fromUnscaledUint(0.5).add(
```

Comment:
In the generalized version, this `0.5` will be a parameter set after deployment, not necessarily 50%.

Reply:
Also, the decimal handling is a bit off, but working on a fix.